### PR TITLE
Add filepath as sourceRoot in SingleFileProject

### LINKED ISSUE #26945: Improve Project API to return valid source root paths for single file projects
**Description:**
Please note that the current project API returns a temporary directory path when trying to access the source root of a `SingleFileProject` instance. Therefore it can be improved so that the project API will return the file path of one of the following. 
- path of the single file
- path of the parent directory of the single file

**Steps to reproduce:**

Try accessing the source root of a single file via project API, as follows.
```ballerina
 String srcRoot = ProjectLoader.loadProject(Paths.get(<FILE_PATH>)).sourceRoot().toAbsolutePath().toString();

```

**Affected Versions:**

**OS, DB, other environment details and versions:**

**Related Issues (optional):**

**Suggested Labels (optional):**

**Suggested Assignees (optional):**

### PR DESCRIPTION
**Pull request by azinneera**
Fixes #26945

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [x] Added necessary tests
  - [x] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

**Review by NipunaRanasinghe — COMMENTED** [review]
Debug server related changes looks good to me. +1

File: cli/ballerina-cli/src/main/java/io/ballerina/cli/task/CreateExecutableTask.java [hunk 1]
```diff
@@ -143,4 +123,30 @@ private void notifyPlugins(Project project, Target target) {
             plugin.codeGenerated(project, target);
         }
     }
+
+    private Path getExecutablePath(Project project) {
+
+        Path fileName = project.sourceRoot().getFileName();
+
+        // If the --output flag is not create the executable in the current directory
```
Comment (Dilhasha) [inline cli/ballerina-cli/src/main/java/io/ballerina/cli/task/CreateExecutableTask.java:131]: // If the --output flag is not **set,** ...

Comment (azinneera) [reply cli/ballerina-cli/src/main/java/io/ballerina/cli/task/CreateExecutableTask.java]: Fixed

**Review by Dilhasha — DISMISSED** [review]

**Review by hevayo — COMMENTED** [review]
Did we check how this affect the diagnostics ?

File: cli/ballerina-cli/src/main/java/io/ballerina/cli/task/CreateExecutableTask.java [hunk 2]
```diff
@@ -38,6 +37,7 @@
 import java.util.ServiceLoader;
 
 import static io.ballerina.cli.launcher.LauncherUtils.createLauncherException;
+import static io.ballerina.cli.utils.FileUtils.geFileNameWithoutExtension;
```
[the remark this answers is not in the document]
Comment (azinneera) [reply cli/ballerina-cli/src/main/java/io/ballerina/cli/task/CreateExecutableTask.java]: Renamed to have the proper name

**Comment by azinneera** [comment]
> Did we check how this affect the diagnostics ?

No, this doesn't affect the diagnostics since we don't use the sourceroot in diagnostics. Wrote a test case to verify this as well.

**Review by hevayo — DISMISSED** [review]

**Review by Dilhasha — APPROVED** [review]

**Pull request merged**